### PR TITLE
A-Boolean Parser

### LINKED ISSUE #89: Update BooleanParser
To accept lower case boolean signals.

### PR DESCRIPTION
**Pull request by sevenseasofbri**
Closes #89  and a part of #88 
Created new interface `TutorialCommand` to parse in particular tutorial commands.
`tutorial circuit` executes `TutorialCircuitCommand`
`tutorial boolean` executes `TutorialBooleanCommand`
Have to update `runtest.bat` with apt commands.

### CODE REVIEW

File: src/main/java/seedu/duke/logic/commands/TutorialCommand.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package seedu.duke.logic.commands;
+
+public interface TutorialCommand {
```
Comment (hughjazzman) [inline src/main/java/seedu/duke/logic/commands/TutorialCommand.java:6]: Interface with only a constant?

File: src/main/java/seedu/duke/model/template/BooleanTemplate.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package seedu.duke.model.template;
 
+import seedu.duke.Duke;
```
Comment (hughjazzman) [inline src/main/java/seedu/duke/model/template/BooleanTemplate.java]: Why is this needed?

File: src/test/java/seedu/duke/logic/parser/ParserTest.java [hunk 3]
```diff
@@ -7,9 +7,10 @@
 import seedu.duke.logic.commands.circuit.AddCircuitCommand;
 import seedu.duke.logic.commands.circuit.CalculateCircuitCommand;
 import seedu.duke.logic.commands.circuit.CircuitCommand;
-import seedu.duke.logic.commands.circuit.HelpCircuitCommand;
 import seedu.duke.logic.commands.circuit.SetCircuitCommand;
 import seedu.duke.logic.commands.circuit.TemplateCircuitCommand;
+import seedu.duke.logic.commands.circuit.TutorialCircuitCommand;
+import seedu.duke.logic.commands.circuit.TutorialCircuitCommand;
```
Comment (hughjazzman) [inline src/test/java/seedu/duke/logic/parser/ParserTest.java]: Duplicate
```suggestion
```

File: text-ui-test/EXPECTED.TXT [hunk 4]
```diff
@@ -7,20 +7,15 @@ Hello from
 \_____|______|_____|_|  \___|\__,_|_|\__| /_/    \_\___/___/_|___/\__\__,_|_| |_|\__|
 What is your name?
 Hello struggling CG1111 student
-You have entered Tutorial Mode!
-Type 'exit' if you want to leave this mode and go back to the application.
-Lets get started! First choose a template you want to work with.You can choose from rc, rl, lc and r templates.
-eg. To select the Resistor-Capacitor circuit template, enter 'template rc.'
+Invalid Command!
```
Comment (hughjazzman) [inline text-ui-test/EXPECTED.TXT]: Should not be invalid command, need to change `input.txt` to use `tutorial circuit` instead.

**Review by hughjazzman — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/model/template/BooleanTemplate.java (hunk 2 above)
Comment (sevenseasofbri) [reply src/main/java/seedu/duke/model/template/BooleanTemplate.java]: Accidently got added cuz intellij optimised the imports lol

File: text-ui-test/EXPECTED.TXT (hunk 4 above)
Comment (sevenseasofbri) [reply text-ui-test/EXPECTED.TXT]: Yup yup had mentioned when i created the PR:
> Have to update runtest.bat with apt commands.

File: src/test/java/seedu/duke/logic/parser/ParserTest.java (hunk 3 above)
Comment (sevenseasofbri) [reply src/test/java/seedu/duke/logic/parser/ParserTest.java]: Lol didn't notice this. Good eye

File: src/main/java/seedu/duke/logic/commands/TutorialCommand.java (hunk 1 above)
Comment (sevenseasofbri) [reply src/main/java/seedu/duke/logic/commands/TutorialCommand.java:6]: Couldn't think of a good way to generalise the tutorial command whilst also splitting between the two kinds of circuits. Since the both the classes already inherited from BooleanCommand/CircuitCommand I thought the only way we could define the characteristics of a TutorialCommand and implement was through an interface (oof)

File: src/main/java/seedu/duke/model/template/BooleanTemplate.java (hunk 2 above)
Comment (sevenseasofbri) [inline src/main/java/seedu/duke/model/template/BooleanTemplate.java]:
```suggestion
```

File: src/main/java/seedu/duke/logic/commands/TutorialCommand.java (hunk 1 above)
Comment (sevenseasofbri) [reply src/main/java/seedu/duke/logic/commands/TutorialCommand.java:6]: @hughjazzman  I might be able to add a function declaration here (startTutorial()), I guess I add it in another PR, or ?

**Review by hughjazzman — APPROVED** [review]
LGTM
